File: RTSlabsCodeExercises.java
public class RTSlabsCodeExercises {

    // O(arr.length) time complexity - O(1) space complexity
    public static String elementsAboveAndBelowN(int[] arr, int n) {
        int above = 0;
        int below = 0;
        
        for (int num : arr) {
            if (num > n) {
                above++;
            } else if (num < n) {
                below++;
            }
        }

        return "above: " + above + ", below: " + below;
    }

    // O(s.length()) time complexity and space complexity
    public static String rotateByN(String s, int n) {

        int rotation = n % s.length();

        if (rotation < 0) rotation += s.length();

        int overFlowPoint = s.length() - rotation;

        String rotatedString = s.substring(overFlowPoint) + s.substring(0, overFlowPoint);

        return rotatedString;

    }

    public static void main(String[] args) {
        int[] arr = {46,9,-23,0,-45,88,12};
        System.out.println(elementsAboveAndBelowN(arr, 0));

        System.out.println(rotateByN("Pittsburgh Steelers", 6));
    }

}

File: python-answers-for-questions-1-and-2/rtsLabsCodeExercises.py
# O(len(arr)) time complexity - 0(1) space complexity
def elementsAboveAndBelowN(arr, n):
    above = 0
    below = 0

    for num in arr:
        if num > n:
            above += 1

        elif num < n:
            below += 1

    print(f"above: {above}, below: {below}")

# O(len(string)) time complexity and space complexity
def rotateByN(string, n):
    
    rotation = n % len(string)

    if rotation < 0:
        rotation += len(string)

    overFlowPoint = len(string) - rotation

    print(string[overFlowPoint:] + string[0:overFlowPoint])

